mnist_cuda_rpc_conv_error/mnist_cuda_rpc_linear.py

```diff
@@ -0,0 +1,115 @@
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from torchvision import datasets, transforms
+import torch.multiprocessing as mp
+import torch.distributed.rpc as rpc
+import os
+import concurrent.futures
+from torch.distributed.optim import DistributedOptimizer
+from torch.distributed.rpc import RRef
+import torch.distributed.autograd as dist_autograd
+from tqdm import tqdm
+from rpc_framework import MyRPCPipeline, MyRPCPipelineWrapper
+
+
+def run_main():
+    batch_size = 100
+
+    transform = transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Normalize((0.1307,), (0.3081,))
+    ])
+
+    train_data = datasets.MNIST('./data', train=True, download=True, transform=transform)
+    valid_data = datasets.MNIST('./data', train=False, transform=transform)
+
+    train_dataloader = torch.utils.data.DataLoader(train_data, batch_size=batch_size)
+    valid_dataloader = torch.utils.data.DataLoader(valid_data, batch_size=batch_size)
+
+    model = MyRPCPipeline(
+        MyRPCPipelineWrapper(nn.Sequential(
+            nn.Flatten(),
+            nn.Linear(28 * 28, 128),
+            nn.ReLU(),
+            nn.Linear(128, 64),
+            nn.ReLU(),
+            nn.Linear(64, 10),
+        ), "worker1:0", rpc_mode='cuda'),
+    )
+
+    criterion = nn.CrossEntropyLoss()
+    optimizer = DistributedOptimizer(
+        optim.Adam,
+        model.parameter_rrefs(),
+    )
+
+    loaders = {"train": train_dataloader, "valid": valid_dataloader}
+
+    max_epochs = 10
+    accuracy = {"train": [], "valid": []}
+    for epoch in range(max_epochs):
+        print(f"Epoch: {epoch+1}")
+        epoch_correct = 0
+        epoch_all = 0
+        for k, dataloader in loaders.items():
+            for i, (x_batch, y_batch) in enumerate(tqdm(dataloader)):
```

Comment:
nit: a potential future optimization is to load the next data using the resource bubble between forward and backward on device 0. This optimization is super-lower priority.